### PR TITLE
feat(Deps/Boost): add support boost

### PR DESCRIPTION
**Pull request by Winfidonarleyan**
- Moved to #3753

### CODE REVIEW

File: apps/installer/includes/os_configs/ubuntu.sh [hunk 1]
```diff
@@ -6,7 +6,7 @@ sudo apt-get update
 if [[ $CONTINUOUS_INTEGRATION ]]; then
   sudo apt-get -y install build-essential libtool make cmake cmake-data clang openssl libgoogle-perftools-dev \
   libssl-dev libmysqlclient-dev libmysql++-dev libreadline6-dev zlib1g-dev libbz2-dev libace-dev mysql-client \
-  libncurses5-dev
+  libncurses5-dev libboost-system1.6*-dev libboost-filesystem1.6*-dev libboost-program-options1.6*-dev libboost-iostreams1.6*-dev
```
Comment (FrancescoBorzi) [inline apps/installer/includes/os_configs/ubuntu.sh:9]: we don't need the `libboost-filesystem1.6*-dev`

File: apps/installer/includes/os_configs/ubuntu.sh [hunk 2]
```diff
@@ -17,7 +17,8 @@ else
     *)
       sudo apt-get install -y git cmake make gcc g++ clang libmysqlclient-dev \
       libssl-dev libbz2-dev libreadline-dev libncurses-dev \
-      mysql-server libace-6.* libace-dev curl unzip
+      mysql-server libace-6.* libace-dev curl unzip \
+      libboost-system1.6*-dev libboost-filesystem1.6*-dev libboost-program-options1.6*-dev libboost-iostreams1.6*-dev
```
Comment (FrancescoBorzi) [inline apps/installer/includes/os_configs/ubuntu.sh:21]: no need for `libboost-filesystem1.6*-dev`

File: deps/boost/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+include (CheckCXXSourceCompiles)
+
+if (WIN32)
+  # On windows the requirements are higher according to the wiki.
+  set(BOOST_REQUIRED_VERSION 1.70)
+else()
+  set(BOOST_REQUIRED_VERSION 1.67)
+endif()
+
+find_package(Boost ${BOOST_REQUIRED_VERSION} REQUIRED system filesystem program_options iostreams)
```
Comment (FrancescoBorzi) [inline deps/boost/CMakeLists.txt:37]: no need for `filesystem`

File: deps/boost/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  INTERFACE
+    -DBOOST_DATE_TIME_NO_LIB
+    -DBOOST_REGEX_NO_LIB
+    -DBOOST_CHRONO_NO_LIB
+    -DBOOST_SERIALIZATION_NO_LIB
+    -DBOOST_CONFIG_SUPPRESS_OUTDATED_MESSAGE
+    -DBOOST_ASIO_NO_DEPRECATED
+    -DBOOST_SYSTEM_USE_UTF8)
+
+if (NOT boost_filesystem_copy_links_without_NO_SCOPED_ENUM)
```
Comment (FrancescoBorzi) [inline deps/boost/CMakeLists.txt:59]: do we need this?

**Comment by Winfidonarleyan** [comment]
#3753

**Pull request closed without merging**